src/constraints/RBEConstraint.C

```diff
@@ -0,0 +1,107 @@
+#include "RBEConstraint.h"
+#include "MooseMesh.h"
+
+registerMooseObject("ProteusApp", RBEConstraint);
+
+InputParameters
+RBEConstraint::validParams()
+{
+  InputParameters params = NodalConstraint::validParams();
+  params.addClassDescription(
+    "Constrains secondary node to move as a linear combination of primary nodes.");
+  params.addRequiredParam<BoundaryName>("primary_node_set",
+    "The boundary ID associated with the primary nodes.");
+  params.addRequiredParam<BoundaryName>("secondary_node_set",
+    "The boundary ID associated with the secondary side.");
+  params.addRequiredParam<Real>("penalty", "The penalty used for the boundary term.");
```

Comment:
Is there anything to add about what range of values are valid for the penalty?

Reply:
No, I think we can assume the user knows to use a positive number.